addons/mail/static/src/models/thread/thread.js

```diff
@@ -1363,6 +1363,46 @@ function factory(dependencies) {
             return [['replace', this.activities.filter(activity => activity.state === 'overdue')]];
         }
 
+        /**
+         * Current partner seen infos.
+         *
+         * @private
+         * @returns {[mail.partner_seen_infos]}
+         */
+        _computeCurrentPartnerSeenInfos() {
+            if (!this.messaging) {
+                return [['unlink-all']];
+            }
+            if (!this.messaging.currentPartner) {
+                return [['unlink-all']];
+            }
+            return [['replace', this.partnerSeenInfos.filter(partnerSeenInfo => partnerSeenInfo.partner === this.messaging.currentPartner)]]
+        }
+
+        /**
+         * Not a real field, used to trigger `markAsFetch` when new message(s) is(are) fetched
+         *
+         * @private
+         */
+        _computeShouldBeSetAsFetched() {
+            if (this.model !== 'mail.channel') {
+                return;
+            }
+            if (this.channel_type === 'channel') {
+                return;
+            }
+            if (!this.mainCacheIsLoaded) {
+                return;
+            }
+            const lastFetchedMessage = this.currentPartnerSeenInfos.length > 0 &&
+                this.currentPartnerSeenInfos[0] &&
+                this.currentPartnerSeenInfos[0].lastFetchedMessage;
+            if (this.lastMessage && (!lastFetchedMessage || this.lastMessage.id > lastFetchedMessage.id))
+            {
+                this.markAsFetched();
```

Comment:
If we do this PR in master we need to also fix `channel_fetched` to give the fetched message id as param, similar to what we did for seen. Otherwise we might have concurrency issue of marking as fetch a message not yet received.

But I think this should be acceptable to fix in 14 without this suggestion.

Reply:
Not sure it's that simple as `channel_fetched` python-side allows bulk call so... that would mean you could mark a group of channel as fetched with the same message. Should we "ensure_one"-ise it ?

Reply:
As a general rule we should try to remove ensure_one and allow batch everywhere, not the other way around.
But anyway we need to rework all our RPCs to use routes instead, so that guest can call them from public chats. So I guess we can do that change then, with route that can take a map of channel/fetched message and handle it in batch as possible.

Reply:
So `def channel_fetched(self, message_ids):` with `message_ids` as a dict `channelId: messageId` ?

Reply:
That would make the method become "api model" instead of using self. It would be one way of doing it.

Another way of doing it is that message_ids must be an array with the same length and order as ids in self.

Reply:
I see, & which approach do you prefer ?